### PR TITLE
Improved scoring

### PR DESCRIPTION
**Pull request by nurgan**
- calculates the traveled distance of the cookie between all hits
- does not count houses that were already hit as scored hits

### CODE REVIEW

File: src/CookiePhysicsComponent.cpp [hunk 1]
```diff
@@ -74,14 +74,18 @@ void CookiePhysicsComponent::updatePhysics(float deltaTime) {
         // The color change has to be done here,
         // due to the bouncing off, the cookie never really hits the object.
         // TODO(nurgan) implement checking for object type and only change color if it is a "house"
-        if( objHit->getRenderComponent()->getShader().compare("Green")) {
+
+
+        cookieState.hits++;
+        cookieState.hitPositions.push_back(holder_->getPosition());
+        //if( objHit->getRenderComponent()->getShader().compare("Green")) {
```
Comment (rgarmsen2295) [inline src/CookiePhysicsComponent.cpp]: I'd say just remove this now

File: src/CookiePhysicsComponent.cpp [hunk 2]
```diff
@@ -115,4 +117,20 @@ float CookiePhysicsComponent::calculateScore() {
...
+            glm::vec3(cookieState.hitPositions[0].x, 0.0, cookieState.hitPositions[0].z) );
+
+    // go through all further hits
+    for(int i = 1; i < cookieState.hitPositions.size(); i++) {
+        distance += glm::distance(
+                glm::vec3(cookieState.hitPositions[i-1].x, 0.0, cookieState.hitPositions[i-1].z),
+                glm::vec3(cookieState.hitPositions[i].x, 0.0, cookieState.hitPositions[i].z) );
+    }
+
+    return distance;
```
Comment (rgarmsen2295) [inline src/CookiePhysicsComponent.cpp:129]: awesome!

**Review by rgarmsen2295 — APPROVED** [review]
Looks good other than minor nit, good stuff!

File: src/CookiePhysicsComponent.cpp (hunk 1 above)
Comment (joncatanio) [reply src/CookiePhysicsComponent.cpp]: Agreed, we can always just check git history if we need to see how it was done.

File: src/GameObject.h [hunk 3]
```diff
@@ -30,6 +30,9 @@ class GameObject {
     // Properties for player moveable objects
     bool toggleMovement;
 
+	// Can a Cookie be delivered to this object
+	bool cookieDeliverable;
```
Comment (joncatanio) [inline src/GameObject.h:34]: I would name this to something more generic. Recall that we will want to have the Grandpa/Baby throw something other than a cookie.

**Review by joncatanio — COMMENTED** [review]
Looks good, just a few comments on naming, other than that I like it.

**Pull request merged**